### PR TITLE
fix: add k8s permissions to fix patch bug

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
We were encountering an issue where k8s would not be able to update pod labels - this ended up not being necessary to solve the problem, but I think the patch pod permission should still be granted to the cluster.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/resourcemanagers/kubernetes_resource_manager.go
Comment (hamidzr) [inline master/internal/resourcemanagers/kubernetes_resource_manager.go:377]: is this intentional?

**Review by hamidzr — APPROVED** [review]

File: master/internal/resourcemanagers/kubernetes_resource_manager.go
Comment (eecsliu) [reply master/internal/resourcemanagers/kubernetes_resource_manager.go:377]: yes - the pod label doesn't allow the "-" character, so this code is pointless.

**Comment by dannysauer** [comment]
We gonna wait until Friday to merge this, so it can be a full year to get a one-word fix in? :stuck_out_tongue:

**Comment by eecsliu** [comment]
absolutely we'll wait until Friday 😂

**Pull request merged**
